src/dqe_idx.erl

```diff
@@ -227,6 +228,28 @@ metrics(Collection) ->
 
 %%--------------------------------------------------------------------
 %% @doc
+%% Returns a list of metric path suffixes of `Depth' that are prefixed
+%% by the given probe `Prefix', which can also be empty.
+%%
+%% For example:
+%% metrics(<<"collection">>, <<>>, 1) -> [<<"base">>].
+%% metrics(<<"collection">>, <<"base">>, 1) -> [<<"cpu">>].
+%% metrics(<<"collection">>, <<>>, 2) -> [[<<"base">>.<<"cpu">>]].
+%%
+%% Actual results are binary-encoded in `dproto' encoding.
+%% @end
+%%--------------------------------------------------------------------
+-spec metrics(Collection::collection(),
+              Prefix::metric(),
+              Depth::pos_integer()) ->
+                    {ok, [Metric::metric()]} |
```

Comment:
That's a good exhaustive list, don't have anything to add.  On our side, any usages of the dqe_idx only update or add to the index and for this we convert lists to binary prior to making the fn call.

I see the advantages of both approaches, but I lean toward binary() for two reasons:

- enforces a form of contractual verification between components i.e. data encoded with dproto should soundly decode on the receiving system/component.  This cannot be mandated with lists, where bad input of [<<"base">>, [[[<<"cpu">>]]]] may have unintended consequences.  It is of course possible to construct an invalid binary as well, but it would not decode properly leading to fast failure.

- conventions used in other parts of the dalmatiner stack, such as `ddb_connection`.  Using `ddb_connection:list(collection(), metric())` expects metric() to be encoded using dproto.  This conversion would need to be done within the dqe_idx_idx in order to communicate with ddb_connection.